transmart-core-db/src/main/groovy/org/transmartproject/db/tree/TreeNode.groovy

```diff
@@ -268,4 +271,20 @@ class TreeNode {
         }
     }
 
+    final String getDimension() {
+        switch (tableName) {
+            case 'concept_dimension':
+                return DimensionImpl.CONCEPT.name
+            case 'patient_dimension':
+                return DimensionImpl.PATIENT.name
+            case 'modifier_dimension':
+                return 'modifier'
```

Comment:
Here it would make sense to query the `dimension_description` table to fetch the dimension name associated with that modifier (the modifier code is in the dimcode).